x/name/keeper/keeper.go

```diff
@@ -89,58 +104,105 @@ func (k Keeper) ResolvesTo(ctx sdk.Context, name string, addr sdk.AccAddress) bo
 // SetNameRecord binds a name to an address.
 func (k Keeper) SetNameRecord(ctx sdk.Context, name string, addr sdk.AccAddress, restrict bool) error {
 	var err error
-	if name, err = k.Normalize(ctx, name); err != nil {
+	normalizedName, err := k.Normalize(ctx, name)
+	if err != nil {
 		return err
 	}
 	if err = types.ValidateAddress(addr); err != nil {
 		return types.ErrInvalidAddress.Wrap(err.Error())
 	}
+	nameKey, err := types.GetNameKeyPrefix(normalizedName)
+	if err != nil {
+		return types.ErrNameNotBound.Wrapf("failed to get name key prefix for name %q: %v", normalizedName, err)
```

Comment:
polish: It's not a prefix here, it's the actual key.

```suggestion
		return types.ErrNameNotBound.Wrapf("failed to get key for name %q: %v", normalizedName, err)
```